scheduler/test/cook/test/tools.clj

```diff
@@ -791,6 +791,27 @@
       (is (= [(make-job 1 2 2048) (make-job 2 1 1024)]
              (util/filter-based-on-user-quota {test-user {:count 4, :cpus 20, :mem 6144}} user->usage queue))))))
 
+(deftest test-filter-pending-jobs-for-quota
+  (let [test-user-1 "john"
+        test-user-2 "bob"
+        user->usage {test-user-1 {:count 1, :cpus 1, :mem 1} test-user-2 {:count 1, :cpus 1, :mem 1}}
+        pool-quota {:count 4, :cpus 100, :mem 100}
+        job-user (atom 0)
+        make-job (fn [user]
+                   (swap! job-user inc)
+                   {:db/id @job-user
+                    :job/user user
+                    :job/resource [{:resource/type :cpus, :resource/amount 1}
+                                   {:resource/type :mem, :resource/amount 1}]})
+        queue [(make-job test-user-1) (make-job test-user-1) (make-job test-user-1) (make-job test-user-2)]
+        user->quota {test-user-1 {:count 2, :cpus 100, :mem 100} test-user-2 {:count 2, :cpus 100, :mem 100}}
+        [job-1 job-2 job-3 job-4] queue]
+    ; If we filtered pool quota first, only the first two jobs in the
+    ; queue would be seen by the user quota and we'd only launch job-1.
```

Comment:
I don't understand this comment -- each of the 4 jobs in the queue is asking for 1 cpu and 1 mem, and the pool quota is 4 jobs / 100 cpus / 100 mem. So why would the pool quota filter to only the first 2 jobs?

Reply:
There's two jobs already running (user->usage)

Reply:
ah, thanks!